packages/flutter/lib/src/widgets/tween_animation_builder.dart

```diff
@@ -0,0 +1,211 @@
+  /// first time.
+  ///
+  /// Any (subclass of) [Tween] is accepted as an argument. For example, to
+  /// animate the height or width of a [Widget], use a [Tween<double>], or
+  /// check out the [ColorTween] to animate the color property of a [Widget].
+  ///
+  /// Any [Tween] provided must have a non-null [Tween.end] value.
+  ///
+  /// ## Ownership
+  ///
+  /// The [TweenAnimationBuilder] takes full ownership of the provided [Tween].
+  /// Once a [Tween] instance has been passed to [TweenAnimationBuilder] its
+  /// properties should not be accessed or changed anymore as they may have been
+  /// modified by the [TweenAnimationBuilder]. If you need to change the
+  /// [Tween], create a **new instance** with the new values.
+  ///
+  /// It is good practice to never store a [Tween] provided to a
+  /// [TweenAnimationBuilder] in an instance variable.
+  final Tween<T> tween;
+
+  /// Called every time the animation value changes.
+  ///
+  /// The current animation value is passed to the builder along with the
+  /// [child]. The builder should build a [Widget] based on the current
+  /// animation value and incorporate the [child] into it, if it is non-null.
+  final TweenAnimationBuilderCallback<T> builder;
+
+  /// The child widget to pass to the builder.
+  ///
+  /// If a builder callback's return value contains a subtree that does not
+  /// depend on the animation, it's more efficient to build that subtree once
+  /// instead of rebuilding it on every animation tick.
+  ///
+  /// If the pre-built subtree is passed as the child parameter, the
+  /// [TweenAnimationBuilder] will pass it back to the [builder] function so
+  /// that it can be incorporated into the build.
+  ///
+  /// Using this pre-built child is entirely optional, but can improve
+  /// performance significantly in some cases and is therefore a good practice.
+  final Widget child;
+
+  /// Called every time the [AnimationStatus] of the underlying animation
+  /// changes.
+  ///
+  /// This can be useful to trigger additional actions (e.g. another animation)
+  /// at the end of the current animation (which is signaled by
+  /// [AnimationStatus.completed]).
+  final AnimationStatusListener animationStatusListener;
+
+  @override
+  ImplicitlyAnimatedWidgetState<ImplicitlyAnimatedWidget> createState() {
+    return _TweenAnimationBuilderState<T>();
+  }
+}
+
+class _TweenAnimationBuilderState<T> extends AnimatedWidgetBaseState<TweenAnimationBuilder<T>> {
+  Tween<T> _currentTween;
+
+  @override
+  void initState() {
+    widget.tween.begin ??= widget.tween.end;
+    _currentTween = widget.tween;
+    super.initState();
+    if (widget.animationStatusListener != null) {
+      controller.addStatusListener(widget.animationStatusListener);
+    }
+    if (_currentTween.begin != _currentTween.end) {
+      controller.forward();
+    }
+  }
+
+  @override
+  void didUpdateWidget(TweenAnimationBuilder<T> oldWidget) {
+    super.didUpdateWidget(oldWidget);
```

Comment:
maybe add a comment saying what the superclass does here (i assume it unconditionally calls forEachTween?)